app/Models/User.php

```diff
@@ -351,4 +358,24 @@ public function scopeWithUsersWhoArentBlockedBy(Builder $query, User $user)
             $query->where('user_id', $user->getKey());
         });
     }
+
+    public function isNotificationAllowed(string $notification_class): bool
+    {
+        if (!empty($this->notifications)) {
+            foreach (Arr::collapse($this->notifications) as $notification_type => $value) {
+                if (NotificationTypes::from($notification_type)->getClass() == $notification_class) {
+                    return false;
+                }
+            }
+        }
+
+        return true;
+    }
+
+    public function notify($instance)
+    {
+        if ($this->isNotificationAllowed($instance::class)) {
+            app(Dispatcher::class)->send($this, $instance);
+        }
+    }
```

Comment:
Instead of doing a global check in the user model it's best that these settings are checked in their respective listeners to keep the model small.

Reply:
What if I take this logic out to a service ? I think that would keep the model and listeners clean.

Reply:
Just a check inside the listeners is fine 👍